### PR TITLE
fix(site): fix template schedule options

### LINKED ISSUE #13082: template schedule settings refactor - fixes
There are a few things we need to modify from https://github.com/coder/coder/pull/13031.

The Dormancy/Failure Cleanup descriptions were removed. We need to return some info on the features.
- [x] [Dormancy Threshold](https://github.com/coder/coder/commit/2f7f9d022a789d4d6061522f28923584550dd008#diff-9c32de4d7ab3847b8bdef6efea784026eadbfab0486b5c1aad70e2d454704d61L512-L515) (Replace text on left side with this)
- [x] [Dormancy Auto Deletion](https://github.com/coder/coder/commit/2f7f9d022a789d4d6061522f28923584550dd008#diff-9c32de4d7ab3847b8bdef6efea784026eadbfab0486b5c1aad70e2d454704d61L545-L548)
- [x] [Failure Cleanup](https://github.com/coder/coder/commit/2f7f9d022a789d4d6061522f28923584550dd008#diff-9c32de4d7ab3847b8bdef6efea784026eadbfab0486b5c1aad70e2d454704d61L484-L486)

image

We can return them under the text for each input box (like they currently are for Autostop requirement) or to the left. 

---

The "Enforce changes across all workspaces" option currently applies the opposite behavior it describes - it should be inverted.

### PR DESCRIPTION
**Pull request by BrunoQuaresma**
https://github.com/coder/coder/assets/3165839/38c06f9f-0a20-432a-987b-5486f574306b

Closes https://github.com/coder/coder/issues/13082

### CODE REVIEW

**Comment by BrunoQuaresma** [comment]
@stirby I'm deploying this PR so you can do some QA before we merge it.

File: site/src/pages/TemplateSettingsPage/TemplateSchedulePage/TemplateScheduleForm.tsx
Comment (stirby) [inline site/src/pages/TemplateSettingsPage/TemplateSchedulePage/TemplateScheduleForm.tsx:429]: I'm okay with reverting the checkbox to match the API. Less confusion for users. 

We need to revert this too:

> By default, workspaces will inherit the Autostop timer from this template. Enabling this option allows users to set custom Autostop timers on their workspaces or turn off the timer.

File: site/src/pages/TemplateSettingsPage/TemplateSchedulePage/TemplateScheduleForm.tsx
Comment (stirby) [inline site/src/pages/TemplateSettingsPage/TemplateSchedulePage/TemplateScheduleForm.tsx:487]: Nice copy here 👍

File: site/src/pages/TemplateSettingsPage/TemplateSchedulePage/TemplateScheduleForm.tsx
Comment (stirby) [inline site/src/pages/TemplateSettingsPage/TemplateSchedulePage/TemplateScheduleForm.tsx:537]: This is good. Can we do text decoration? If it's a small lift, would like to make the following **bold**:

> Once a workspace is deleted it cannot be recovered.

If it's more than a one-line change we don't have to worry about it.

**Review by stirby — COMMENTED** [review]
Left some comments on the option descriptions. Will approve after testing but should also get @aslilac's approval

File: site/src/pages/TemplateSettingsPage/TemplateSchedulePage/TemplateScheduleForm.tsx
Comment (BrunoQuaresma) [reply site/src/pages/TemplateSettingsPage/TemplateSchedulePage/TemplateScheduleForm.tsx:537]: What about making it bold + a warning color? 

Screenshot 2024-04-26 at 12 53 32

**Review by stirby — APPROVED** [review]
Perfect

**Review by aslilac — APPROVED** [review]
sorry, left a bunch of comments on friday and then didn't submit 😵‍💫

File: site/src/pages/TemplateSettingsPage/TemplateSchedulePage/TemplateScheduleForm.tsx
Comment (stirby) [reply site/src/pages/TemplateSettingsPage/TemplateSchedulePage/TemplateScheduleForm.tsx:429]: I couldn't find the thread but agree with the sentiment. As long as we keep it uniform.

File: site/src/pages/TemplateSettingsPage/TemplateSchedulePage/TemplateScheduleForm.tsx
Comment (stirby) [reply site/src/pages/TemplateSettingsPage/TemplateSchedulePage/TemplateScheduleForm.tsx:487]: I agree we should cut it.

File: site/src/components/StackLabel/StackLabel.tsx
Comment (BrunoQuaresma) [reply site/src/components/StackLabel/StackLabel.tsx:12]: `StackProps` does not work very well because of the`ForwardProps`.

Comment (BrunoQuaresma) [reply site/src/components/StackLabel/StackLabel.tsx:12]: Screenshot 2024-04-29 at 14 13 06

**Pull request merged**
